File: Arrays/searching-algorithms/linear-search/linear.cpp
#include <iostream>
#include <vector>

// Returns the index of target if found, else -1
int linearSearch(const std::vector<int>& arr, int target) {
    for (size_t i = 0; i < arr.size(); ++i) {
        if (arr[i] == target) {
            return static_cast<int>(i);
        }
    }
    return -1;
}

int main() {
    std::vector<int> arr = {10, 23, 45, 7, 89, 12};
    int target = 89;
    int result = linearSearch(arr, target);
    
    if (result != -1)
        std::cout << "Element found at index: " << result << "\n";
    else
        std::cout << "Element not found.\n";
    
    return 0;
}

File: sorting-searching/searching-algorithms/linear-search/linear.py
from typing import List

def linear_search(arr: List[int], target: int) -> int:
    """Returns the index of target if found, else -1."""
    for i, val in enumerate(arr):
        if val == target:
            return i
    return -1

if __name__ == "__main__":
    arr = [10, 23, 45, 7, 89, 12]
    target = 89
    result = linear_search(arr, target)
    
    if result != -1:
        print(f"Element found at index: {result}")
    else:
        print("Element not found.")